sigstore-java/src/main/java/dev/sigstore/rekor/v2/client/RekorV2Verifier.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2025 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.rekor.v2.client;
+
+import com.google.common.hash.Hashing;
+import com.google.protobuf.ByteString;
+import dev.sigstore.encryption.signers.Verifiers;
+import dev.sigstore.merkle.InclusionProofVerificationException;
+import dev.sigstore.merkle.InclusionProofVerifier;
+import dev.sigstore.proto.rekor.v1.TransparencyLogEntry;
+import dev.sigstore.rekor.client.Checkpoints;
+import dev.sigstore.rekor.client.RekorEntry.Checkpoint;
+import dev.sigstore.rekor.client.RekorParseException;
+import dev.sigstore.rekor.client.RekorVerificationException;
+import dev.sigstore.trustroot.SigstoreTrustedRoot;
+import dev.sigstore.trustroot.TransparencyLog;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+
+/* Verifier for rekor v2 entries. */
+public class RekorV2Verifier {
+  private final List<TransparencyLog> tlogs;
+
+  public static RekorV2Verifier newRekorV2Verifier(SigstoreTrustedRoot trustRoot) {
+    return newRekorV2Verifier(trustRoot.getTLogs());
+  }
+
+  public static RekorV2Verifier newRekorV2Verifier(List<TransparencyLog> tlogs) {
+    return new RekorV2Verifier(tlogs);
+  }
+
+  private RekorV2Verifier(List<TransparencyLog> tlogs) {
+    this.tlogs = tlogs;
+  }
+
+  public void verifyEntry(TransparencyLogEntry entry, Instant timestamp)
+      throws RekorVerificationException {
+    if (entry.getInclusionProof() == null) {
+      throw new RekorVerificationException("No inclusion proof in entry.");
+    }
+
+    var tlog =
+        TransparencyLog.find(tlogs, entry.getLogId().getKeyId().toByteArray(), timestamp)
```

Comment:
were we trying to use checkpoint id here instead or something? @haydentherapper ?

Reply:
I think I was too optimistic about switching over to checkpoint key ID quickly and deprecating log ID. Using log ID is fine still, we will populate both in the future.